storage.tf

```diff
@@ -1,32 +1,74 @@
 # CICD utilities Storage
 
-resource "google_storage_bucket" "cicd_utils" {
-  name                        = "${var.infra_name}-cicd-utils"
+resource "random_string" "deployment_utils_bucket_suffix" {
+  length           = 4
+  special          = false
+  lower            = true
+  upper            = false
+}
+
+resource "google_storage_bucket" "deployment_utils" {
+  name                        = "${var.infra_name}-utils-${random_string.deployment_utils_bucket_suffix.result}"
   location                    = "US"
   storage_class               = "STANDARD"
   uniform_bucket_level_access = true
   provider                    = google.offensive-pipeline
 }
 
+
 resource "google_storage_bucket_object" "disable_windows_defender_ps" {
   name   = "scripts/Powershell/disabledefender.ps1"
-  bucket = google_storage_bucket.cicd_utils.name
+  bucket = google_storage_bucket.deployment_utils.name
   source = "${path.module}/scripts/Powershell/disabledefender.ps1"
 }
 
 
-# Gitlab instance deploy storage
-
-resource "google_storage_bucket" "gitlab_deploy_utils" {
-  name                        = "${var.infra_name}-gitlab-deploy-utils"
-  location                    = "US"
-  storage_class               = "STANDARD"
-  uniform_bucket_level_access = true
-  provider                    = google.offensive-pipeline
-}
-
 resource "google_storage_bucket_object" "gitlab_install_script" {
   name   = "scripts/bash/gitlab_install.sh"
-  bucket = google_storage_bucket.gitlab_deploy_utils.name
+  bucket = google_storage_bucket.deployment_utils.name
   source = "${path.module}/scripts/bash/gitlab_install.sh"
-}
\ No newline at end of file
+}
+
+# Customized Gitlab backup script
+
+resource "google_storage_bucket_object" "gitlab_backup_script" {
+  name   = "scripts/bash/gitlab_backup.sh"
+  bucket = google_storage_bucket.deployment_utils.name
+  source = "${path.module}/scripts/bash/gitlab_backup.sh"
+}
+
+resource "google_storage_bucket_object" "gitlab_backup_script_exec" {
+  name   = "scripts/bash/gitlab_backup_exec.sh"
+  bucket = google_storage_bucket.deployment_utils.name
+  source = "${path.module}/scripts/bash/gitlab_backup_exec.sh"
+}
+
+# Migration resource
+
+resource "google_storage_bucket_object" "gitlab_migrate_script" {
+  count  = var.migrate_gitlab ? 1 : 0
```

Comment:
I would remove it from here, you don't mind uploading this script to storage bucket.
Will be used only when needed.